### DISCUSSION TITLE
Chain * does not support contract "ensUniversalResolver" error/warning

### DISCUSSION THREAD

**Discussion by alexnguyennz** [wevm/viem#781] [General]
Hi can this be made less intense in a future release? It prints out a massive error block in console. Thanks.

**Comment by lekevicius** [comment]
Usually a couple. I don't want to put in logic just to handle Hardhat / anvil - silently failing is ok.

**Reply by xh3b4sd** [reply]
Wouldn't it make sense to have a chain object have unsupported contracts configured? Then the e.g. Hardhat chain object could set to not support contract `ensUniversalResolver` and we could implement and configure a quite reasonable workflow. If a chain object explicitly says that it does not support a contract, it should not be invoked and an error should not be escalated.

**Comment by shrugs** [comment]
i haven't monkeypatched a built-in since 2011 but here we are

```ts
"use client";

import { useIsomorphicLayoutEffect } from "usehooks-ts";

// https://github.com/wagmi-dev/viem/discussions/781
export function useShutTheFuckUpAboutENS() {
  useIsomorphicLayoutEffect(() => {
    const orig = window.console.error;
    window.console.error = function (...args) {
      if (args[0]?.name === "ChainDoesNotSupportContract") return;
      orig.apply(window.console, args);
    };
  }, []);
}

```

**Comment by mentasuave01** [comment]
it's very annoying

**Comment by jmrossy** [comment]
+1 for graceful failure here instead of this wall of text, which can print several times if a Dapp is multi-chain:
<img width="1258" alt="Screenshot 2023-12-14 at 12 43 16 PM" src="https://github.com/wevm/viem/assets/710751/59684f7e-f4e8-4280-9af2-c8695359ba66">

**Reply by tmm** [reply]
Mind posting an issue with a minimal reproduction? `Chain "Ethereum"` (assuming you are using `mainnet`) should already have this contract attached to it.

**Reply by jmrossy** [reply]
@tmm Thanks for the response!

Here's a minimal stackblitz that can repro. The error will show once you connect to Metamask: https://stackblitz.com/edit/new-wagmi-nu8kvs?file=src%2Fwagmi.ts

The reason this is happening for Ethereum mainnet is presumably because my custom chain config doesn't specify an `ensRegistry` address. While that's fixable for mainnet, I work with a lot of newer/smaller L1s that don't have a de-facto ENS instance. Would it be possible to avoid exercising the ENS lookup when chain configs don't specify an ens registry address? Though I guess that's more of a wagmi-related concern than Viem.
